filter.ts

```diff
@@ -15,7 +15,11 @@ export function matchFilter(
   filter: Filter,
   event: Event & {id: string}
 ): boolean {
-  if (filter.ids && filter.ids.indexOf(event.id) === -1) return false
+  if (filter.ids && filter.ids.indexOf(event.id) === -1) {
+    if (!filter.ids.some(prefix => prefix.length < 64 && event.id.startsWith(prefix))) {
+      return false
+    }
+  }
   if (filter.kinds && filter.kinds.indexOf(event.kind) === -1) return false
   if (filter.authors && filter.authors.indexOf(event.pubkey) === -1)
```

Comment:
shouldn't we also allow prefixes here?

Reply:
I agree, ids and authors https://github.com/nostr-protocol/nips/blob/master/01.md

Reply:
maybe it would also make sense in "#e" and "#p", but that is not in nip-01 so out of scope for this PR.